packages/typescript-estree/src/convert.ts

```diff
@@ -822,11 +822,20 @@ export class Converter {
 
       // Exceptions
 
-      case SyntaxKind.ThrowStatement:
+      case SyntaxKind.ThrowStatement: {
+        const { expression } = node;
+        if (
+          expression.kind === SyntaxKind.Identifier &&
+          !(expression as ts.Identifier).text
+        ) {
+          throw createError(this.ast, node.end, 'Expression expected.');
+        }
+
```

Comment:
```suggestion
        if (ts.isIdentifier(node.expression) && !node.expression.text) {
          throw createError(this.ast, node.end, 'Expression expected.');
        }
```

Reply:
i'm kidna wandering if we can't move this check to `SyntaxKind.Identifier` and just throw this error for it when `node.text` is '' ?